blue_hardware_interface/src/blue_hardware_interface.cpp

```diff
@@ -68,9 +78,88 @@ BlueHW::BlueHW(ros::NodeHandle &nh) : nh_(nh) {
       this);
 }
 
+void BlueHW::doSwitch(const std::list<hardware_interface::ControllerInfo>& start_list,
+                      const std::list<hardware_interface::ControllerInfo>& stop_list) {
+
+  std::vector<std::string> joint_hardware_effort_controllers{
+    "effort_controllers/JointTrajectoryController",
+    "effort_controllers/JointGroupEffortController",
+    "blue_controllers/BlueJointGroupPositionController",
+    "blue_controllers/BlueJointGroupCTC",
+  };
+  std::vector<std::string> joint_hardware_position_controllers{
+    "position_controllers/JointGroupPositionController",
+  };
+
+  std::vector<std::string> gripper_hardware_effort_controllers{
+    "effort_controllers/GripperActionController",
+    "effort_controllers/JointGroupEffortController",
+  };
+  std::vector<std::string> gripper_hardware_position_controllers{
+  };
+
+  // Acquire the kinematics lock
+  std::lock_guard<std::mutex> lock(motor_driver_mutex_);
```

Comment:
maybe move this mutex into `BLDCDriver`? it'd make the critical section much smaller

(we can also save this for a future PR)

Reply:
Ya lets push this for a future PR